modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -119,6 +147,100 @@ public void testAddCustomObjectField() throws Exception {
 					objectFieldNameException.getMessage());
 			}
 		}
+
+		ObjectFieldBuilder objectFieldBuilder = new ObjectFieldBuilder();
+
+		try {
+			_testAddCustomObjectField(
+				objectFieldBuilder.businessType(
+					ObjectFieldConstants.BUSINESS_TYPE_TEXT
+				).dbType(
+					ObjectFieldConstants.DB_TYPE_STRING
+				).defaultValue(
+					RandomTestUtil.randomString()
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"a" + RandomTestUtil.randomString()
+				).objectFieldSettings(
+					_getObjectFieldSettings(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT)
+				).build());
+
+			Assert.fail();
+		}
+		catch (ObjectFieldDefaultValueException
+					objectFieldDefaultValueException) {
+
+			Assert.assertEquals(
+				StringBundler.concat(
+					"Object field can only have a default type when the ",
+					"business type is \"",
+					ObjectFieldConstants.BUSINESS_TYPE_PICKLIST, "\""),
+				objectFieldDefaultValueException.getMessage());
+		}
+
+		String defaultValue = RandomTestUtil.randomString();
+
+		try {
+			_testAddCustomObjectField(
+				objectFieldBuilder.businessType(
+					ObjectFieldConstants.BUSINESS_TYPE_PICKLIST
+				).listTypeDefinitionId(
+					_listTypeDefinition.getListTypeDefinitionId()
+				).defaultValue(
+					defaultValue
+				).build());
+
+			Assert.fail();
+		}
+		catch (ObjectFieldDefaultValueException
+					objectFieldDefaultValueException) {
+
+			Assert.assertEquals(
+				StringBundler.concat(
+					"Default value \"", defaultValue, "\" is not a list entry ",
+					"in list definition ",
+					_listTypeDefinition.getListTypeDefinitionId()),
+				objectFieldDefaultValueException.getMessage());
+		}
+
+		try {
+			_testAddCustomObjectField(
+				objectFieldBuilder.businessType(
+					ObjectFieldConstants.BUSINESS_TYPE_PICKLIST
+				).defaultValue(
+					"open"
+				).build());
+
+			Assert.fail();
+		}
+		catch (ObjectFieldStateException objectFieldStateException) {
+			Assert.assertEquals(
+				"Object field default value can only be set when the " +
+					"picklist is a state",
+				objectFieldStateException.getMessage());
+		}
+
+		try {
+			_testAddCustomObjectField(
+				objectFieldBuilder.state(
+					true
+				).build());
+
+			Assert.fail();
+		}
+		catch (ObjectFieldStateException objectFieldStateException) {
+			Assert.assertEquals(
+				"Object field must be required when the state is true",
+				objectFieldStateException.getMessage());
+		}
```

Comment:
Hi @seltong,

Lets try to apply something like [this method](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectViewLocalServiceTest.java#L400-L432), I believe the that case is equals to this.